src/utils/misc.ts

```diff
@@ -79,3 +79,27 @@ export const createPlaceholderToken = (tokenData: Token): TokenView => {
   };
   return createToken({ tokenData, userTokenData, allowancesMap: {} });
 };
+
+interface GetSelectedSellTokenProps {
+  sellTokensOptions: TokenView[];
+  selectedSellTokenAddress: string | undefined;
+  nativeCurrency: string;
+  selectedTokenMap: { [address: string]: Token };
+}
+
+export const getSelectedSellToken = ({
```

Comment:
I think this is better, but probably could go closer to where it's used as it's not really a util, my suggestion is to move this to

`src/client/components/app/Transactions/getSelectedSellToken.ts`

nit: Also some new lines in this function would improve code readability.